service/storage_service.cc

```diff
@@ -985,6 +985,20 @@ class topology_coordinator {
         }
     }
 
+    future<std::pair<node_to_work_on, bool>> global_token_metadata_barrier(node_to_work_on&& node) {
+        bool exec_command_res;
+        std::tie(node, exec_command_res) = co_await exec_global_command(std::move(node),
+            raft_topology_cmd { raft_topology_cmd::command::barrier_and_drain },
+            true);
+        if (!exec_command_res) {
+            co_return std::make_pair(std::move(node), exec_command_res);
```

Comment:
nit: `std::make_pair` is not necessary in new C++ versions, it was a workaround for a language defect, today we can use the constructor directly (`std::pair{...}`).